botocore/client.py

```diff
@@ -82,9 +82,30 @@ def create_client(self, service_name, region_name, is_secure=True,
         service_name = first_non_none_response(responses, default=service_name)
         service_model = self._load_service_model(service_name, api_version)
         cls = self._create_client_class(service_name, service_model)
+        if region_name is not None:
```

Comment:
It might be worth breaking out the region normalization into it's own function but we can always refactor that later.